source/features/table-input.tsx

```diff
@@ -0,0 +1,79 @@
+import './table-input.css';
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import TableIcon from 'octicon/diff-added.svg';
+import * as pageDetect from 'github-url-detection';
+import * as textFieldEdit from 'text-field-edit';
+
+import features from '.';
+
+function generateGfmTable([w, h]: [number, number]): string {
+	return `${'|     '.repeat(w)}|\n${'|-----'.repeat(w)}|\n${`${'|     '.repeat(w)}|\n`.repeat(h)}`.slice(0, -1);
+}
+
+function addTable(event: delegate.Event<MouseEvent, HTMLButtonElement>): void {
+	event.preventDefault();
+	(event.delegateTarget.parentElement!.parentElement as HTMLDetailsElement)!.open = false;
+
+	const field = event.delegateTarget.form!.querySelector('textarea')!;
+	const endsWithNl = field.value.length && !field.value.endsWith('\n');
+	const cursorPos = field.value.length + (endsWithNl ? 4 : 3);
+
+	textFieldEdit.insert(field, (endsWithNl ? '\n' : '') + generateGfmTable(getSquarePos(event.delegateTarget)));
+
+	field.focus();
+	field.setSelectionRange(cursorPos, cursorPos);
+}
+
+function highlightSquares(event: delegate.Event<MouseEvent, HTMLButtonElement>): void {
+	const [x, y] = getSquarePos(event.delegateTarget);
+	[...event.delegateTarget.parentElement!.children].forEach(child => {
+		const [cx, cy] = getSquarePos(child as HTMLElement);
+		child.classList.toggle('rgh-add-table-popup__square--selected', cx <= x && cy <= y);
+	});
+}
```

Comment:
If you follow all of my suggestions, this should look like:
```suggestion
function highlightSquares(event: delegate.Event<MouseEvent, HTMLButtonElement>): void {
	const hover = event.delegateTarget;
	for (const cell of event.delegateTarget.parentElement!.children) {
		cell.classList.toggle('selected', cell.dataset.x <= hover.dataset.x && cell.dataset.y <= hover.dataset.y)
	}
}
```